### PR TITLE
Fixing timestamps and steps in metric logging

### PR DESCRIPTION
**Pull request by mrkaye97**
Getting rid of the ability for users to submit their own steps and timestamps when logging metrics in favor of `lightMLFlow` just doing it for them. This makes life a lot easier for everyone

### CODE REVIEW

File: R/rest.R [hunk 1]
```diff
@@ -114,7 +114,7 @@ call_mlflow_api <- function(..., client, query = NULL, data = NULL, verb = "GET"
           data,
           pretty = TRUE,
           auto_unbox = TRUE,
-          na = NULL
+          na = "null"
```
Comment (tonyelhabr) [inline R/rest.R:117]: ah this is probably my bad. i think i had added the `na = NULL`

Comment (mrkaye97) [reply R/rest.R:117]: all good, I don't think this actually affected anything and tbh it's weird that your approach didn't work

File: R/runs.R [hunk 2]
```diff
@@ -40,34 +40,86 @@ assert_new_col_length <- function(x, metrics) {
...
+#' @importFrom purrr imap_int
+#' @importFrom magrittr add
 #'
 #' @export
-log_metrics <- function(..., timestamp = NA, step = NA, run_id = get_active_run_id(), client = mlflow_client()) {
+log_metrics <- function(..., run_id = get_active_run_id(), client = mlflow_client()) {
 
   metrics <- get_key_value_df(...)
 
-  assert_new_col_length(timestamp, metrics)
```
Comment (tonyelhabr) [inline R/runs.R:64]: so we can get rid of `assert_new_col_length` now, right?

Comment (tonyelhabr) [reply R/runs.R:64]: it's in some tests, so if we want to keep it, maybe just define it in the test file?

**Review by tonyelhabr — APPROVED** [review]
timestamp stuff looks like a headache. glad you found a solution

File: R/runs.R (hunk 2 above)
Comment (mrkaye97) [reply R/runs.R:64]: Oh yeah lemme check

**Pull request merged**